src/Discord/Internal/Types/Prelude.hs

```diff
@@ -82,40 +84,28 @@ epochTime = posixSecondsToUTCTime 0
 
 type ColorInteger = Integer
 
-makeTable :: (Data t, Enum t) => t -> [(Int, t)]
-makeTable t = map (\cData -> let c = fromConstr cData in (fromEnum c, c)) (dataTypeConstrs $ dataTypeOf t)
+class Data a => InternalDiscordType a where
+  discordTypeStartValue :: a
+  fromDiscordType :: a -> Int
+  discordTypeTable :: [(Int, a)]
+  discordTypeTable =  map (\d -> (fromDiscordType d, d)) (makeTable discordTypeStartValue)
+    where 
+      makeTable :: Data b => b -> [b]
+      makeTable t = map fromConstr (dataTypeConstrs $ dataTypeOf t)
```

Comment:
This feels weird. Is `discordTypeStartValue` necessary?

Reply:
`Data` is a bit weird, and to get the constructors you need an example of the value, I think.